From: Kurt Garloff <scs@garloff.de>
Subject: Disable all availability zone hints for networking resources

Many OpenStack clouds do not use network availability zones.
Unfortunately, still passing hints will result in API errors.
Thus we need to patch them out (short of making terraform clever
enough to detect it).

Co-authored-by: Mathias Fechner <mathias.fechner@plusserver.com>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

--- neutron.tf	2020-11-05 05:20:18.670617087 +0100
+++ neutron.tf.patch	2020-11-05 05:24:34.377334696 +0100
@@ -172,7 +172,7 @@ resource "openstack_compute_secgroup_v2"
 
 resource "openstack_networking_network_v2" "net_management" {
   name                    = "net-${var.prefix}-management"
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_subnet_v2" "subnet_management" {
@@ -190,7 +190,7 @@ resource "openstack_networking_subnet_v2
 
 resource "openstack_networking_network_v2" "net_internal" {
   name                    = "net-${var.prefix}-internal"
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_subnet_v2" "subnet_internal" {
@@ -209,7 +209,7 @@ resource "openstack_networking_subnet_v2
 
 resource "openstack_networking_network_v2" "net_provider" {
   name                    = "net-${var.prefix}-provider"
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_subnet_v2" "subnet_provider" {
@@ -228,7 +228,7 @@ resource "openstack_networking_subnet_v2
 
 resource "openstack_networking_network_v2" "net_external" {
   name                    = "net-${var.prefix}-external"
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_subnet_v2" "subnet_external" {
@@ -265,7 +265,7 @@ resource "openstack_networking_port_v2"
 
 resource "openstack_networking_network_v2" "net_storage_frontend" {
   name                    = "net-${var.prefix}-storage-frontend"
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_subnet_v2" "subnet_storage_frontend" {
@@ -284,7 +284,7 @@ resource "openstack_networking_subnet_v2
 
 resource "openstack_networking_network_v2" "net_storage_backend" {
   name                    = "net-${var.prefix}-storage-backend"
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_subnet_v2" "subnet_storage_backend" {
@@ -308,7 +308,7 @@ data "openstack_networking_network_v2" "
 resource "openstack_networking_router_v2" "router" {
   name                    = var.prefix
   external_network_id     = data.openstack_networking_network_v2.public.id
-  availability_zone_hints = [var.network_availability_zone]
+  # availability_zone_hints = [var.network_availability_zone]
 }
 
 resource "openstack_networking_router_interface_v2" "router_interface" {
